### ISSUE TITLE
[stable/owncloud] Owncloud health checks, ingress, .htaccess

### ISSUE THREAD

**Issue by ajb** [helm/charts#978]
Hi there,

I'm verrrry new to k8s and helm, and I've been playing around with installing the Owncloud chart on a new GKE cluster. First of all, let me say that I'm extremely impressed! :heart: to all the contributors for making this such a great experience.

I've hit a couple hiccups that I'll enumerate here. I'm not sure that any of them are technically bugs, but they required workarounds that weren't obvious to a beginner, for whatever that's worth :). Or maybe they're just misunderstandings, in which case I'd appreciate any help!

- The chart's default ServiceType is `LoadBalancer`, which on GKE, spins up a new Load Balancer. However, there doesn't seem to be an option to add TLS to this load balancer.  So in order to add TLS, I needed to create an ingress, but in order to do that, I had to specify `serviceType=NodePort`. It took a while to figure out why the ingress wouldn't work with `serviceType=LoadBalancer` (see https://github.com/kubernetes/kubernetes/issues/24041)

- The `readinessProbe` is set as a curl command, `curl --fail localhost/status.php`. However, this type of readinessProbe is not supported by GKE's ingress, which only supports a HTTPS path AFAICT. (Seems related to https://github.com/kubernetes/kubernetes/issues/20555)

- Apache doesn't recognize the `.htaccess` file in Owncloud's data directory, which means that all uploads are publicly accessible via the internet. 😱  (I opened an issue here: https://github.com/bitnami/bitnami-docker-owncloud/issues/53)

These issues all arose from trying to get (what I think is) a reasonably standard configuration deployed for my personal use. If I'm misunderstanding the intent of this chart and trying to do something that's way out of bounds, please let me know :smile:

**Comment by cassioconti** [comment]
Like your second bullet, I'm also interested in the GKE to support health check other them HTTP(S). Although this looks more as a Google issue than Kubernetes.

**Comment by DylanGraham** [comment]
Did you ever get this working with TLS?
I'm having the same problem.

**Comment by ericuldall** [comment]
The LoadBalancer created by a GKE Service is an Google L4 (TCP/UDP) Load Balancer. There is no way for this type of LB to handle SSL termination. You can still expose port 443 from your L4 Load Balancer and let your application handle SSL termination.

Also, Ingress should still work when using a LoadBalanced service. We do this in almost all of our applications.

**Comment by ustuehler** [comment]
I can confirm that the /data directory, and every file in it, is *still* word-readable in the latest version of the stable/owncloud chart, or rather the latest version of the `bitnami/owncloud` image  (https://github.com/bitnami/bitnami-docker-owncloud/issues/53). If this isn't getting addressed at all, I would rather advocate to remove that chart temporarily, than letting users leak data like this.

I tried to understand how the `nami` tool actually generates the config files, but gave up in order to just f*ing plug that hole somehow, for now. My workaround is a new entrypoint script that appends the following to `/bitnami/apache/conf/vhosts/htaccess/owncloud-htaccess.conf`, if not present:

```
# Fix publicly readable data directory
<Directory "/opt/bitnami/owncloud/data">
  AllowOverride All
</Directory>
```

**Comment by ustuehler** [comment]
I can confirm that https://github.com/bitnami/bitnami-docker-owncloud/issues/53 is solved in version 0.5.9 of this [chart](https://github.com/kubernetes/charts/tree/master/stable/owncloud). When upgrading from previous versions, `AllowOverride None` should be changed to `AllowOverride All` in `/bitnami/apache/conf/vhosts/owncloud-*.conf`, and the `Include` line can be removed.
